src/components/calculator.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+function Calculator() {
+  const renderButton = (text, name, color, style) => (
+    <div className="flex-1 px-2 py-2 justify.cnter flex items-center text-white text-2xl font-semibold">
+      <button
+        type="button"
+        className={`rounded-full h-12 w-20 flex items-center ${color} justify-center shadow-lg hover:border-gray-500 focus:outline-none ${style}`}
+        name={name}
+      >
+        {text}
+      </button>
+    </div>
+  );
+
+  return (
+    <div className="mx-auto overflow-hidden mt-10 shadow-lg mb-2 bg-[#a4a4a4] shadow-lg border rounded-lg lg:w-2/6 mg:w-3/6 sm:w-4/6">
+      <div className="">
+        <div className="p-5 text-white text-right text-3xl bg-[#515151]">
+          <span className="text-white-500">
+            0
+          </span>
+        </div>
+
+        <div className="flex items-stretch bg-[#c8c8c8] h-20">
+          {renderButton('7', '7', 'bg-gray-800')}
+          {renderButton('8', '8', 'bg-gray-800')}
+          {renderButton('9', '9', 'bg-gray-800')}
+          {renderButton('x', '*', 'bg-orange-500')}
+        </div>
+
+        <div className="flex items-stretch bg-[#c8c8c8] h-20">
+          {renderButton('4', '4', 'bg-gray-800')}
+          {renderButton('5', '5', 'bg-gray-800')}
+          {renderButton('6', '6', 'bg-gray-800')}
+          {renderButton('-', '-', 'bg-orange-500')}
+        </div>
+
+        <div className="flex items-stretch bg-[#c8c8c8] h-20 mb-4">
+          {renderButton('1', '1', 'bg-gray-800')}
+          {renderButton('2', '2', 'bg-gray-800')}
+          {renderButton('3', '3', 'bg-gray-800')}
+          {renderButton('+', '+', 'bg-gray-800')}
+        </div>
+
+        <div className="flex items-stretch bg-[#c8c8c8] h-20 mb-4">
+          {renderButton('0', '0', 'bg-gray-800', 'w-52')}
+          {renderButton('.', '.', 'bg-gray-800')}
+          {renderButton('=', '=', 'bg-gray-800')}
+        </div>
+      </div>
+    </div>
+  );
+}
```

Comment:
- [ ] Kindly ensure the Calculator component has a child component. I see you are trying to implement that using the `renderButton` function. But `renderButton` in this case is not really a component. It is just a function within the Calculator, the way one could have a `handleSubmit` method within a component.

  Components have their own files and are imported for use. For example, the Button component, as you have it is a component, but `renderButton` isn't.

- [ ] While creativity and flair are good traits, for this very project we are required to follow the design given instead of using ours. Kindly modify the design to match the one given. This is your design at the moment.

  ![image](https://github.com/Dorcas126/math-magicians/assets/81468675/d5b11d89-8696-43a9-8ba2-d6d3cda0ce72)